java/hello-world/README.md

```diff
@@ -3,20 +3,132 @@
 This is a simple application that demonstrates using the native HBase API
 to connect to and interact with Cloud Bigtable.
 
-## Provision a cluster
+See the [documentation for this
+sample](https://cloud.google.com/bigtable/docs/samples-java-hello) for a brief
+explanation of the code.
 
-Follow the instructions in the [user documentation](https://cloud.google.com/bigtable/docs/creating-cluster)
-to create a Google Cloud Platform project and Cloud Bigtable cluster if necessary.
-You'll need to reference your project id, zone and cluster id to run the application.
 
-## Run the application
+## Downloading the sample
 
-First, set your [Google Application Default Credentials](https://developers.google.com/identity/protocols/application-default-credentials)
-and [install Maven](http://maven.apache.org/guides/getting-started/maven-in-five-minutes.html) if necessary.
+Download the sample app and navigate into the app directory:
 
+1.  Clone the [Cloud Bigtable examples repository][github-repo], to your local
+    machine:
 
-    $ mvn package
-    $ mvn exec:java -Dbigtable.projectID=<myProject> -Dbigtable.clusterID=<myCluster> -Dbigtable.zone=<myZone>
+        git clone https://github.com/GoogleCloudPlatform/cloud-bigtable-examples.git
+
+    Alternatively, you can [download the sample][github-zip] as a zip file and
+    extract it.
+
+2.  Change to the Hello World code sample directory.
+
+        cd cloud-bigtable-examples/java/hello-world
+
+[github-repo]: https://github.com/GoogleCloudPlatform/cloud-bigtable-examples
+[github-zip]: https://github.com/GoogleCloudPlatform/cloud-bigtable-examples/archive/master.zip
+
+
+## Costs
+
+This sample uses billable components of Cloud Platform, including:
+
++   Google Cloud Bigtable
+
+Use the [Pricing Calculator][bigtable-pricing] to generate a cost estimate
+based on your projected usage.  New Cloud Platform users might be eligible for
+a [free trial][free-trial].
+
+[bigtable-pricing]: https://cloud.google.com/products/calculator/#id=1eb47664-13a2-4be1-9d16-6722902a7572
+[free-trial]: https://cloud.google.com/free-trial
+
+
+## Before you begin
+
+This sample assumes you have [Java 8][java8] installed.
+
+[java8]: http://www.oracle.com/technetwork/java/javase/downloads/
+
+### Installing Maven
+
+These samples use the [Apache Maven][maven] build system. Before getting
```

Comment:
I've found that the version of Maven is too far out-of-date when using `apt-get install`. We don't mention it here: https://cloud.google.com/appengine/docs/java/tools/maven#installing_maven_31